spec/unit/provider/package/pear_spec.rb

```diff
@@ -33,46 +33,46 @@
     end
 
     it 'ignores malformed lines' do
-      described_class.expects(:pear).
+      allow(described_class).to receive(:pear).
         with('list', '-a').
-        returns 'aaa2.1'
-      Puppet.expects(:warning).with('Could not match aaa2.1')
+        and_return 'aaa2.1'
+      allow(Puppet).to receive(:warning).with('Could not match aaa2.1')
       expect(described_class.instances).to eq []
     end
   end
 
   describe '#install' do
     it 'installs a package' do
-      described_class.expects(:pear).
+      allow(described_class).to receive(:pear).
         with('-D', 'auto_discover=1', 'upgrade', '--alldeps', 'dummy')
       provider.install
```

Comment:
To really replace this and keep the test, you should add a `have_received` assertion. This is repeated so I won't comment it everywhere.
```suggestion
      provider.install
      expect(described_class).to have_receive(:pear)
```

Reply:
I already tried this, but it did not work as expected:
* https://travis-ci.org/github/voxpupuli/puppet-php/jobs/672461881#L1193